test/plugin/example_plugin_test.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func ReadNLines(reader io.Reader, n int) ([]string, error) {
+	var lines []string
+	reReader := bufio.NewReader(reader)
+	for i := 0; i < n; i++ {
+		line, err := reReader.ReadString('\n')
+		lines = append(lines, line)
```

Comment:
Maybe we don't care about this in tests but I think if you're appending a lot of lines you might want to go with `string.Builder`.

Reply:
We care about the lines individually so I’m not sure if Builder is appropriate for this case

Reply:
We care about the lines individually so I’m not sure if Builder is appropriate for this case

Reply:
I believe what you want here is [`bufio.Scanner`](https://godoc.org/bufio#Scanner)

Reply:
Used `bufio.Scanner`